plugin/aws/ec2InstanceConsoleOutput.go

```diff
@@ -50,16 +50,16 @@ type consoleOutput struct {
 	content []byte
 }
 
-func (cl *ec2InstanceConsoleOutput) cachedConsoleOutput(ctx context.Context) (consoleOutput, error) {
-	output, err := plugin.CachedOp(ctx, "ConsoleOutput", cl, 30*time.Second, func() (interface{}, error) {
+func (inst *ec2Instance) cachedConsoleOutput(ctx context.Context) (consoleOutput, error) {
+	output, err := plugin.CachedOp(ctx, "ConsoleOutput", inst, 30*time.Second, func() (interface{}, error) {
 		request := &ec2Client.GetConsoleOutputInput{
-			InstanceId: awsSDK.String(cl.inst.id),
+			InstanceId: awsSDK.String(inst.id),
 		}
-		if cl.latest {
-			request.Latest = awsSDK.Bool(cl.latest)
+		if inst.hasLatestConsoleOutput {
```

Comment:
I think this needs to be an argument you pass in. `newEC2InstanceConsoleOutput` is used to generate two different files, to cover `console.out` and `console-latest.out`. From your use in `ec2Instance` we can choose which one, but for use in `ec2InstanceConsoleOutput`'s `Open` function we need to use whatever it was created with (`cl.latest`).

Reply:
You need to essentially preserve the old behavior. Which means add a 2nd argument to `cachedConsoleOutput` that is `latest`. Then when you call it from `ec2Instance` pass `inst.hasLatestConsoleOutput`, when you call it from `ec2InstanceConsoleOutput` pass `cl.latest`.

Reply:
Ah, thank you. I just couldn't make that leap. 

In the meantime, I got to thinking maybe the cachedConsoloeOutput function could get at `latest` if I simply point to the struct that contains it like this:
cl := &ec2InstanceConsoleOutput{}
That's based on the assumption that value for `latest` will always be what it should be at that point. It seems to work. But then I saw you post and suspect maybe this approach isn't correct (e.g. I'm making wrong assumptions). If not though, is there an advantage to doing it one way or the other?

Reply:
I think you're missing what it was doing before. https://github.com/puppetlabs/wash/blob/0.4.0/plugin/aws/ec2Instance.go#L117-L130 creates 2 versions of ec2InstanceConsoleOutput. One passes in `latest=false`, the other - if `hasLatestConsoleOutput == true` - passes in `latest=true`. So we have versions that request with and without console latest.

@ekinanp I don't actually remember why we want the `!latest` version when `hasLatestConsoleOutput == true`. Do you remember why you included it?

Reply:
That might be me mis-interpreting the API docs. https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-console.html implies that they could be two different outputs. If I'm wrong and they're the same output. then I don't think we need to include latest.

Reply:
I went ahead and pushed the changes just now per Michael's earlier guidance. That is, in case we want to merge still keeping the two-versions interpretation. Happy to work on the task of making it one version if that's the way to go... either post-merge or just continuing to update this PR in-place.